railties/lib/rails/generators/rails/controller/controller_generator.rb

```diff
@@ -44,7 +44,8 @@ def generate_routing_code(action)
 
           # Create route
           #     get 'baz/index'
-          route = indent(%{  get '#{file_name}/#{action}'\n}, depth * 2)
+          leading_spaces = depth.zero? ? "" : "  "
+          route = indent(%{#{leading_spaces}get '#{file_name}/#{action}'\n}, depth * 2)
```

Comment:
Since need two indents for each namespace, I think it is correct to multiplied by 2. 
The reason why we adjust the space when `depth`is 0, add a space at the beginning in `route` method.
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/actions.rb#L242 

If depth is not 0, this adjustment is done when adding namespace. 
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/controller/controller_generator.rb#L42